scripts/dss_v2_apdapter.py

```diff
@@ -0,0 +1,489 @@
+            'schema_type': 'link_bundle',
+            'schema_version': '2.0.0',
+            'links': [],
+        }
+        # process links
+        link: JSON
+        for link in old_links_json['links']:
+            new_link = {
+                'link_type': 'process_link',
+                'process_id': link['process'],
+                'process_type': schema_types[link['process']],
+                'inputs': [],
+                'outputs': [],
+                'protocols': [],
+            }
+            for prop in ('input', 'output', 'protocol'):
+                for value in link[f'{prop}s']:
+                    if isinstance(value, str):  # a uuid
+                        new_link[f'{prop}s'].append(
+                            {
+                                f'{prop}_type': schema_types[value],
+                                f'{prop}_id': value,
+                            }
+                        )
+                    elif isinstance(value, dict):
+                        new_link[f'{prop}s'].append(
+                            {
+                                f'{prop}_type': value[f'{prop}_type'],
+                                f'{prop}_id': value[f'{prop}_id'],
+                            }
+                        )
+                    else:
+                        raise Exception(f"Unknown value type in links.json {prop} field.")
+            new_links_json['links'].append(new_link)
+        # supplementary file links
+        if supplementary_files:
+            new_link = {
+                'link_type': 'supplementary_file_link',
+                'entity': {
+                    'entity_type': 'project',
+                    'entity_id': project_uuid
+                },
+                'supplementary_files': [
+                    {'file_id': uuid, 'file_type': 'supplementary_file'}
+                    for uuid in supplementary_files
+                ],
+            }
+            new_links_json['links'].append(new_link)
+
+        return new_links_json
+
+    def _download_blob_as_string(self, blob_key: str) -> bytes:
+        """
+        Perform a file download from the source bucket.
+        """
+        return self.src_bucket.blob(f'blobs/{blob_key}').download_as_string()
+
+    def _upload_file_contents(self, file_contents: JSON, new_name: str, content_type: str):
+        """
+        Perform an upload of JSON data to a file in the destination bucket.
+        """
+        # log.info('Uploading to: %s', new_name)
+        self.dst_bucket.blob(new_name).upload_from_string(data=json.dumps(file_contents, indent=4),
+                                                          content_type=content_type)
+
+    def _copy_file(self, src_blob: Blob, new_name: str):
+        """
+        Perform a bucket to bucket copy of a file.
+        """
+        # log.info('Copying to: %s', new_name)
+        dst_blob = self.src_bucket.copy_blob(blob=src_blob,
```

Comment:
High priority: This seems different to what was discussed on https://github.com/DataBiosphere/azul/issues/1780.